app/models/list.rb

```diff
@@ -0,0 +1,4 @@
+class List < ActiveRecord::Base
+  belongs_to :user
+  has_one :user
```

Comment:
Only one of these relationships with `User` is needed.

As `User` is the `has_one` side of the `User <---> List` relationship, then `List` ought to be the `belongs_to` side. So delete the `has_one :user` line from `list.rb`.